ibis/examples/__init__.py

```diff
@@ -78,11 +78,9 @@ def fetch(
                     if pyarrow.types.is_null(field.type):
                         table = table.set_column(i, field.name, table[i].cast("string"))
 
-            # TODO: It should be possible to avoid this memtable call, once all
-            # backends support passing a `pyarrow.Table` to `create_table`
-            # directly.
-            obj = ibis.memtable(table)
-            return backend.create_table(table_name, obj, temp=True, overwrite=True)
+            obj = ibis.memtable(table, name=table_name)
+            backend._register_in_memory_tables(obj)
+            return backend.table(table_name)
```

Comment:
Won't this drop `obj` (the memtable) at the end of the function, causing it to be collected and the table removed?

Reply:
ARG, yes

Reply:
What if I immediately call `.cache()`?

Reply:
I wonder if we should just allow `temp=True` datafusion. Seems like a lot of effort just to disable a behavior that wasn't so bad.

Reply:
`memtable` seems like currently the only sane way to do that.

Reply:
I'll push up the `cache` "solution" and we can discuss further.

Reply:
> That might have weird consequences for the ddl work @ncclementi is doing? That said, I do think that an engine like datafusion (where all tables are kinda temporary anyway) ignoring temp completely does make sense in isolation.

I know this is closed, but for completion of the discussion this wouldn't affect the work itself. My only concern would be if someone creates a "temporary" and then tries to do a `con.ddl.list_temp_tables()` it will get a not implemented error, because the `create_table` swallows the `TEMPORARY` and such table shows a `BASE_TABLE` in the information_schema, meaning that it will and is listed when `con.ddl.list_tables()` is invoked. 

That being said, there is an upstream issue to raise when `TEMPORARY` is used in the SQL which will cause the same problem down the road.